pkg/cmd/start/main.go

```diff
@@ -57,7 +60,7 @@ func NewStartCommand() *cobra.Command {
 	cmd.Flags().String("openshift-oauth-proxy-image", "openshift/oauth-proxy:latest", "The Docker image location definition for the OpenShift OAuth Proxy")
 	viper.BindPFlag("openshift-oauth-proxy-image", cmd.Flags().Lookup("openshift-oauth-proxy-image"))
 
-	cmd.Flags().String("platform", "kubernetes", "The target platform the operator will run. Possible values: 'kubernetes' and 'openshift'")
+	cmd.Flags().String("platform", "auto-detect", "The target platform the operator will run. Possible values: 'kubernetes' and 'openshift'")
```

Comment:
do we need to keep this flag at all? I would rather remove it altogether and rely on auto detection.

Reply:
We can add it once we need it.

Reply:
If our auto-detect logic fails for some reason in some strange situations, it's good to provide a fallback to the user. So, let's keep it.